grammar.js

```diff
@@ -1734,10 +1734,12 @@ module.exports = grammar({
 
     character_literal: $ => seq(
       '\'',
-      choice(token.immediate(/[^'\\]/), $.escape_sequence),
+      choice($.character_literal_content, $.escape_sequence),
       '\'',
     ),
 
+    character_literal_content: $ => token.immediate(/[^'\\]/),
+
```

Comment:
```suggestion
      choice(
        alias(token.immediate(/[^'\\]/), $.character_literal_content),
        $.escape_sequence
      ),
      '\'',
    ),
```

Reply:
Why is this version preferable? And won't it have the same issue that we discussed [here](https://github.com/tree-sitter/tree-sitter-c-sharp/pull/333#discussion_r1586422465)?

Reply:
ah, right. it's just a little less noisy (since it's not reused anywhere else) and is sorta often done in other grammars. it's not a big difference imo, so this is fine